brainiak/reprsimil/brsa.py

```diff
@@ -85,113 +272,288 @@ class BRSA(BaseEstimator):
+    GP_inten: boolean. Defualt: False
+        Whether to include a kernel defined over the intensity of image.
         GP_space should be True as well if you want to use this,
         because the smoothness should be primarily in space.
-        Smoothness in intensity is just complementary.
-    tol: tolerance parameter passed to the minimizer.
-    verbose : boolean, default: False
-        Verbose mode flag.
-    eta: a small number added to the diagonal element of the
-        covariance matrix in the Gaussian Process prior. This is
-        to ensure that the matrix is invertible.
-    space_smooth_range: the distance (in unit the same as what
+        Smoothness in intensity is just complementary. The idea
+        behind this option is that voxels should have similar
+        SNRs when they are both adjacent (imposed by GP_space)
+        and are of the same tissue type (when their image intensities
+        are close). If you accept the second assumption, then
+        you can set GP_inten as True and provide an array to the `inten`
+        variable, expressing the intensities (brightness) for each voxel.
+    space_smooth_range: float. Default: None
+        The distance (in unit the same as what
         you would use when supplying the spatial coordiates of
         each voxel, typically millimeter) which you believe is
         the maximum range of the length scale parameter of
         Gaussian Process defined over voxel location. This is
         used to impose a half-Cauchy prior on the length scale.
         If not provided, the program will set it to half of the
         maximum distance between all voxels.
-    inten_smooth_range: the difference in image intensity which
+    inten_smooth_range: float. Default: None
+        The difference in image intensity which
         you believe is the maximum range of plausible length
         scale for the Gaussian Process defined over image
         intensity. Length scales larger than this are allowed,
         but will be penalized. If not supplied, this parameter
         will be set to half of the maximal intensity difference.
-    tau_range: the reasonable range of the standard deviation
-        of the Gaussian Process. Since the Gaussian Process is
-        imposed on the log(SNR), this range should not be too
-        large. 10 is a pretty loose range. This parameter is
-        used in a half-Cauchy prior on the standard deviation
-    init_iter: how many initial iterations to fit the model
+    tau_range: float. Default: 5.0
+        The reasonable range of the standard deviation
+        of log(SNR). This range should not be too
+        large. 5 is a loose range.
+        When a Gaussian Process is imposed on the log(SNR),
+        this parameter is used in a half-Cauchy prior
+        on the standard deviation, or an inverse-Gamma prior
+        on the variance of the GP.
+    tau2_prior: callable[[float, int, float], [float, float]],
```

Comment:
Thanks @mihaic I forgot how I wrote this line. It looks like it only takes [float, int, float]. So I will delete the second part of the line as well.